add .ssf for 2019_Jeong_InnerEurasia

nevrome opened the pull request:
The poseidon_IDs column is completely empty. Package and .ssf file are not linked.

93Boy commented:
Could not match Poseidon_IDs into any data fields in .ssf files

AyGhal commented:
Okay. I checked the paper, supplementary info and PANDORA: TU45 is BOT14 and BKZ001 is BOT2016 (Table 1 of the paper). From PANDORA TU44/TU45 are the same individual. And my guess would be that BKZ001 and BKZ002 are also same individual. @stschiff can you double check with Choongwon? 
The rest are modern and the bam files are not shared.

nevrome commented:
Stephan wrote Choongwon and got a helpful reply. My understanding is now that the `poseidon_IDs column should be filled as follows:


| ... | sample_alias | poseidon_IDs | ... |
|--------|--------|--------|--------|
| ... | TU44 | BOT14 | ... |
| ... | TU45 | BOT14 | ... |
| ... | BKZ001.C0101 | BOT2016 | ... |
| ... | BKZ002.A0101 | BOT2016 | ... |
| ... | TU44.merged | BOT14 | ... |
| ... | TU45.merged | BOT14 | ... |

Please confirm this @AyGhal or @stschiff. As Choongwon explicitly writes that `TU44.merged` and `TU45.merged` are per-library merged BAM files that include both screening shotgun and 1240K capture data, and that these versions should be preferably used, I wonder if we should remove the first two rows of the .ssf file with `TU44` and `TU45`. @TCLamnidis would want it like that for minotaur, right?

stschiff commented:
Yes, your table also matches my understanding of the situation. Regarding the merged data, I agree that we should in that case then remove the unmarked ones, as we don't want duplicate reads. @TCLamnidis ?

stschiff commented:
Discussed briefly on Jan 12. Dana's on it.

stschiff commented:
OK, validation fails, for multiple reasons. (Running `trident validate` locally confirms this): Your SSF file is comma-separated, not tab-separated, and it has quotations. Please fix, @93Boy, thanks.

93Boy commented:
This commit has passed the validation test

nevrome commented:
Looks good to me. Can you increase the version number and update the changelog, @93Boy?

93Boy commented:
I have updated the changelog with trident rectify and a fullGeno validation. Now there is a consistency issue. But I have not replaced or overwritten any file in the directory. Do you know why this issue keep happening?

nevrome commented:
So - the problem here is that the package was changed on the master branch since this branch was created. This can happen with branches that are open for a long time. You can prevent it, by merging the current state of the master into the old branch before making additional changes.

It's too late for that now and you instead have to fix the resulting merge conflict. That is also pretty easy, though. GitHub has a neat interface for it. You can click the `Resolve conflicts` button to activate it.

On the left you will see a list of files that has conflicts. And then each file has one or multiple blocks of code where the conflict arises. The first one you'll see is the conflict in `CHANGELOG.txt`:

```
<<<<<<< ssfFor2019_Jeong_InnerEurasia
V 2.0.2: Linked ENA data with poseidon_IDs
V 2.0.1: Genetic_Source_Accession_IDs added
V 2.0.0: updated the package to Poseidon v2.5 and adjusted the .janno file accordingly
V 1.1.0: Added the many modern reference samples to the package
=======
- V 3.0.1: fixed Genotype_Ploidy information
- V 3.0.0: Changed two group names that are incompatible with trident ('<' -> 'lt')
- V 2.1.0: Update of references in the .bib file
- V 2.0.1: Genetic_Source_Accession_IDs added
- V 2.0.0: updated the package to Poseidon v2.5 and adjusted the .janno file accordingly
- V 1.1.0: Added the many modern reference samples to the package
>>>>>>> master
```
It's pretty clear that the `master` branch version on the bottom has additional changes over the `ssfFor2019_Jeong_InnerEurasia` version on the top. To resolve the conflict you can manually combine them. In the process you should also remove the lines Git adds to highlight the merge conflict.

```
- V 3.1.0: added an .ssf file
- V 3.0.1: fixed Genotype_Ploidy information
- V 3.0.0: Changed two group names that are incompatible with trident ('<' -> 'lt')
- V 2.1.0: Update of references in the .bib file
- V 2.0.1: Genetic_Source_Accession_IDs added
- V 2.0.0: updated the package to Poseidon v2.5 and adjusted the .janno file accordingly
- V 1.1.0: Added the many modern reference samples to the package
```

Note that I also changed the version number and log message to better reflect what this change actually does.

When you entered this new version in the code window for `CHANGELOG.txt` you can press the button in the top right corner: `Mark as resolved`. And then you can move on to the `POSEIDON.yml` file and do the same. Here a suitable solution may look like this:

```
poseidonVersion: 2.5.0
title: 2019_Jeong_InnerEurasia
description: The genetic history of admixture across inner Eurasia. Originally AADR
  v42.4.
contributor:
- name: Ayshin Ghalichi
  email: ghalichi@shh.mpg.de
packageVersion: 3.1.0
lastModified: 2024-01-25
genotypeData:
  format: PLINK
  genoFile: 2019_Jeong_InnerEurasia.bed
  genoFileChkSum: 7e8207019df974e2a5a9d6c555958bc9
  snpFile: 2019_Jeong_InnerEurasia.bim
  snpFileChkSum: 71c40efb829c18b7149996d951097e45
  indFile: 2019_Jeong_InnerEurasia.fam
  indFileChkSum: 5d7ffa6f67daa37e706c6d1f6cfd2211
  snpSet: 1240K
jannoFile: 2019_Jeong_InnerEurasia.janno
jannoFileChkSum: ???
sequencingSourceFile: ENAtable.ssf
sequencingSourceFileChkSum: a3787a6b657fcbe368b51f3997b84046
bibFile: 2019_Jeong_InnerEurasia.bib
bibFileChkSum: 1f951edb2083205a1b67061ac0b61d78
changelogFile: CHANGELOG.txt
```

We run into a bit of an issue here, because we don't know the new checksum for the .janno file after the merge. I suggest you complete the merge by clicking `Mark as resolved` also for .janno when you made these changes, and then pull the changes locally to adjust the checksum once more. This should then complete the whole process.

93Boy commented:
@stschiff @nevrome I have updated Poseidon.yml as we discussed at the last Poseidon meeting. But still, the same conflict occurs.

nevrome commented:
The solution I provided above should still work. Can you not just go to `Resolve conflicts` and replace the text in `POSEIDON.yml` there with that? What are you struggling with specifically?

GitHub has some good documentation on merge conflicts [here](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/about-merge-conflicts) and [here](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/resolving-a-merge-conflict-on-github).

TCLamnidis commented:
Sorry for the delay in my reply. If all reads of TU45 are indeed included within TU45.merged, then the first two rows should indeed be removed from the SSF.

stschiff commented:
Thiseas, please get this ready for merging into the main branch, thanks. Inform @AyGhal when done.

TCLamnidis commented:
@AyGhal SSF should be good to go now. 
I also added the `udg` and `library_built` columns needed by Minotaur here. Based on those, I also updated the `udg` field for the two ancient samples in the janno.

AyGhal reviewed (approved):
Looks good to me. Thank you!
